Docs: Variables docs improvements (#125282)
grafana/grafana: docs/sources/visualizations/dashboards/variables/variable-syntax.md

    - enterprise
    - oss
title: Variable syntax
description: Use variable syntax and format options to control how Grafana interpolates dashboard variables in queries, panel titles, and expressions.
weight: 400
---

# Variable syntax

Panel titles and metric queries can refer to variables using two different syntaxes:

- `$varname`
  This syntax is easy to read, but it doesn't allow you to use a variable in the middle of a word.
  **Example:** apps.frontend.$server.requests.count
- `${var_name}` Use this syntax when you want to interpolate a variable in the middle of an expression.
- `${var_name:<format>}` This format gives you more control over how Grafana interpolates values. Refer to [Advanced variable format options](#advanced-variable-format-options) for more detail on all the formatting types.
- `[[varname]]` Do not use. Deprecated syntax, that will be removed in a future release.

Before queries are sent to your data source, the query is _interpolated_, meaning the variable is replaced with its current value.
During interpolation, the variable value might be _escaped_ to conform to the syntax of the query language and where it's used.

> For example, a variable used in a regular expression in an InfluxDB or Prometheus query is regex escaped.

Author: ```suggestion
For example, variables in InfluxDB or Prometheus regular expression queries are automatically escaped.
```
